File: backend/app/schemas/comment.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field, field_validator


class CommentCreate(BaseModel):
    """The comment body is the only client-controlled field."""

    model_config = ConfigDict(extra="forbid")

    body: str = Field(min_length=1, max_length=2000)

    @field_validator("body", mode="before")
    @classmethod
    def _strip_whitespace(cls, value: object) -> object:
        return value.strip() if isinstance(value, str) else value


class CommentResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    author_name: str
    author_avatar_url: str | None
    body: str
    created_at: datetime
    updated_at: datetime


class CommentCreateResponse(BaseModel):
    comment: CommentResponse
    comment_count: int

File: lib/features/community/data/comment.dart
class Comment {
  const Comment({
    required this.id,
    required this.authorName,
    required this.authorAvatarUrl,
    required this.body,
    required this.createdAt,
    required this.updatedAt,
  });

  final String id;
  final String authorName;
  final String? authorAvatarUrl;
  final String body;
  final DateTime createdAt;
  final DateTime updatedAt;

  factory Comment.fromJson(Map<String, dynamic> json) => Comment(
    id: json['id'] as String,
    authorName: json['author_name'] as String,
    authorAvatarUrl: json['author_avatar_url'] as String?,
    body: json['body'] as String,
    createdAt: DateTime.parse(json['created_at'] as String),
    updatedAt: DateTime.parse(json['updated_at'] as String),
  );
}

class CreatedComment {
  const CreatedComment({required this.comment, required this.commentCount});

  final Comment comment;
  final int commentCount;

  factory CreatedComment.fromJson(Map<String, dynamic> json) => CreatedComment(
    comment: Comment.fromJson(json['comment'] as Map<String, dynamic>),
    commentCount: json['comment_count'] as int,
  );
}
